### PR TITLE
Added archive type to metadata

### PR DESCRIPTION
**Pull request by abdallahbeshi**

### CODE REVIEW

**Review by marektresnak — APPROVED** [review]
I guess we could check that the value is one of the allowed values when parsing metadata. But that would be just adding more logic to the checker and wouldn't bring much value.

File: src/Program.cs [hunk 1]
```diff
@@ -55,7 +54,7 @@ private static byte[] ComputeSignature(Archive archive)
                 reportedValue.Value.ToSignatureString(),
                 archive.Metadata.Created.ToSignatureString(),
                 archive.Metadata.TerminalIdentification,
-                operationName,
+                archive.Metadata.ArchiveType,
```
Comment (marektresnak) [inline src/Program.cs:57]: Just realized this doesn't work for v1. It needs to be done in a backwards compatible way.

**Review by marektresnak — CHANGES_REQUESTED** [review]

File: src/Model/ArchiveMetadata.cs [hunk 2]
```diff
@@ -23,6 +24,8 @@ private ArchiveMetadata(string terminalIdentification, IOption<Signature> previo
 
         public ArchiveVersion Version { get; }
 
+        public string ArchiveType { get; }
```
Comment (marektresnak) [inline src/Model/ArchiveMetadata.cs:27]: Since we now have the typed model, this should be an option.

File: src/Program.cs [hunk 3]
```diff
@@ -44,18 +44,21 @@ private static byte[] ComputeSignature(Archive archive)
...
             var taxSummary = archive.TaxSummary;
             var reportedValue = archive.ReportedValue;
-            var operationName = "ARCHIVING";
             var previousSignatureFlag = archive.Metadata.PreviousRecordSignature.Match(
                 _ => "Y",
                 _ => "N"
             );
+            var archiveType = archive.Metadata.Version.Match(
+                ArchiveVersion.v100, _ => "ARCHIVING",
+                ArchiveVersion.v400, _ => archive.Metadata.ArchiveType.GetOrElse("")
```
Comment (marektresnak) [inline src/Program.cs:53]: The whole app is written in a way that invalid data lead to a different failure than the actual signature check failure. I would follow this principle here as well and make sure model metadata doesn't even get created if it's missing some information.

**Review by KaliCZ — APPROVED** [review]

**Pull request merged**